guides/micronaut-data-mongodb/java/src/main/java/example/micronaut/domain/Course.java

```diff
@@ -0,0 +1,78 @@
+package example.micronaut.domain;
+
+import io.micronaut.data.annotation.GeneratedValue;
+import io.micronaut.data.annotation.Id;
+import io.micronaut.data.annotation.MappedEntity;
+import io.micronaut.data.annotation.Relation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@MappedEntity("courses")
+public class Course {
+
+    @Id
+    @GeneratedValue
+    private String id;
+
+    private String name;
+
+    @Relation(value = Relation.Kind.MANY_TO_MANY, mappedBy = "courses")
+    private List<Student> students;
+
+    public Course(String name) {
+        this.name = name;
+        this.students = new ArrayList<>();
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public List<Student> getStudents() {
```

Comment:
when the relationship is not fetched it can be null, thus annotate it as `@Nullable`